### ISSUE TITLE
Archive of FB video and reels is not playing on https://replayweb.page/

### ISSUE THREAD

**Issue by tsemachh** [webrecorder/archiveweb.page#204]
**Recreate the bug:**
Archive from the chrome extension in facebook either:
1. Reel - like [https://www.facebook.com/reel/261061716437457](https://www.facebook.com/reel/261061716437457) 
2. Video - like [https://www.facebook.com/watch/?v=10225216681907826&ref=sharing](https://www.facebook.com/watch/?v=10225216681907826&ref=sharing)

Do not try to replay them on the extension because of extension [replay crash chrome bug](https://github.com/webrecorder/archiveweb.page/issues/194)
Instead download them as WACZ and replay them on the [https://replayweb.page/](https://replayweb.page/)

**Expected results:**
Videos are playing good.
**Actual results:**
Both videos are not playable.

Attaching few hints why it's not working:
[bgvideorecord.txt](https://github.com/webrecorder/archiveweb.page/files/13853959/bgvideorecord.txt)
[fbvideoconsole.txt](https://github.com/webrecorder/archiveweb.page/files/13853960/fbvideoconsole.txt)
[fbvideoreplayconsole.txt](https://github.com/webrecorder/archiveweb.page/files/13853958/fbvideoreplayconsole.txt)

[bgreelrecord.txt](https://github.com/webrecorder/archiveweb.page/files/13853961/bgreelrecord.txt)
The reel replay is just like the video replay so I guess it's same issue.

I think we have some recursion on the redirect at:(But I am not sure :-))
```
            async getReplayResponse(t, e) {
                let r = this.getCanonRedirect(t);
                if (r)
                    return r;
                const i = {
                    pageId: t.pageId
                };
                r = await this.store.getResource(t, this.prefix, e, i);
                const {request: n, url: s} = t;
                if (("video" === n.destination || "audio" === n.destination) && "navigate" !== n.mode)
                    for (; r && r.status >= 301 && r.status < 400; ) {
                        const n = new URL(r.headers.get("location"),s);
                        t.url = n.href,
                        console.log(`resolve redirect ${s} -> ${t.url}`),
                        r = await this.store.getResource(t, this.prefix, e, i)
                    }
                return r
            }
```

**Comment by tareefdev** [comment]
I have the same issue, some archived FB posts/reels work while others are not.

**Comment by julescarbon** [comment]
I'm having the same issue playing back videos from Facebook. I noticed some nonstandard behavior with how the video is requested. Normally with HTML5 video, the MP4 would be requested in chunks with the `Range` header. Instead, the MP4 requests have `bytestart` and `byteend` in the URL query string.

I was able to manually reconstruct the video from a WACZ containing a single Facebook post, using `warcio.ArchiveIterator` to extract the individual requests, then reassembling the MP4s using the byte ranges.  Archiving a single Facebook video resulted in multiple MP4s being downloaded.  Two of them were the "content" from the Facebook post. The video had trouble seeking, similar to how it behaved when played from the WACZ, but I was able to make a working file using ffmpeg to make a new container (trivially `ffmpeg -i file.mp4 -c copy fixed.mp4`). I then used ffmpeg again to combine the working streams into one file. Some things to note:

- The video and audio streams are in separate MP4s.
- Facebook does not necessarily request an entire video, and there are often chunks are overlapping/missing. (This may be normal.)
- Facebook also requests portions of other videos, which do not play and were not recovered.  Of the five MP4s in my WACZ, two were content, two were small/empty, and the other was somewhat large but did not contain any playable streams.

**Comment by ikreymer** [comment]
The 0.12.6 release includes significant improvements to Facebook capture, includes videos and reels, though haven't tested those specifically. @julescarbon your analysis is correct, but there's a way to force facebook to serve full MP4s and use range requests which I believe should now be enabled in 0.12.6 - no guarantee how long that will work. Earlier approaches were parsing and fixing the DASH manifests that FB uses, which is another technique that could be tried later..

**Comment by ikreymer** [comment]
I believe this should mostly be fixed in 0.12.6, closing for now!
